### PR TITLE
feat(pm): surface own-PR Greptile review via notification-blind detector

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Problem

When Bob opens his own PR, neither the PR creation nor subsequent Greptile/Codecov bot reviews generate `review_requested` or `mention` notifications. Additionally, `check_greptile_scores` intentionally seeds (doesn't emit) on first discovery to avoid flooding when a new repo is onboarded. Combined, this means a brand-new PR's first Greptile review is silently recorded and only re-emitted after the 1-hour cooldown window — by which time a manual intervention is often needed anyway.

**Concrete case**: `gptme/gptme#2860` received a Greptile 3/5 review and was invisible to project-monitoring for ~1h. The existing cooldown-based re-emit eventually fired, but the detection gap is unnecessary.

## Solution

Add `check_own_pr_review_state()` to `activity-gate.sh`:

- **Zero extra API cost**: reads Greptile state files written by `check_greptile_scores` (must run after it)
- **Emits on first discovery** of a changed `(head_sha, greptile_score, merge_state)` signature — no seed-only behavior
- **Routes correctly**:
  - `greptile < 4` → `greptile_needs_fix`
  - `greptile = 4` → `greptile_needs_improvement`
  - `greptile >= 5` + `CLEAN` → skip (`check_merge_ready` already handles this)
  - `DIRTY`/`CONFLICTING` → skip (`check_merge_conflicts` handles)
  - `BLOCKED`/`UNKNOWN` → skip (CI not yet settled)
- **Anti-repeat**: one dispatch per signature; re-dispatches only when the signature changes (new commit, re-review, or merge-state change)

## What existing primitives can't do

`check_greptile_scores` seeds on first discovery by design (onboarding noise prevention). There's no way to ask it "emit on first discovery for own-PRs only" without forking its logic. This function is a thin reader over its state files, not a duplicate of its API-calling logic.

## Testing

Manually verified on the `activity-gate.sh` shellcheck + bash syntax check pass:
```bash
bash -n scripts/github/activity-gate.sh  # syntax OK
shellcheck scripts/github/activity-gate.sh  # no errors
```

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
Addressed the Greptile finding in `16122e8`.

`check_own_pr_review_state()` now treats any `Greptile >= 5` result as non-actionable, regardless of merge state, so `BEHIND` / `UNSTABLE` / similar states no longer emit a fake `greptile_needs_improvement` item for a perfect review.

Added a regression test covering the `5/5 + BEHIND` case:
- `uv run pytest tests/test_activity_gate_own_pr_review.py -q`
- `uv run pytest tests/test_activity_gate_cache.py -q`

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: scripts/github/activity-gate.sh [hunk 1]
```diff
@@ -694,6 +694,94 @@ check_greptile_scores() {
...
+
+        # Read Greptile score from state file written by check_greptile_scores
+        local greptile_state_file="$STATE_DIR/${repo_safe}-pr-${pr_number}-greptile.state"
+        local greptile_score=""
+        if [ -f "$greptile_state_file" ]; then
+            greptile_score=$(cut -d: -f1 < "$greptile_state_file")
+        fi
+
+        # No Greptile review on file yet — skip
+        [ -z "$greptile_score" ] || [ "$greptile_score" = "null" ] && continue
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply scripts/github/activity-gate.sh]: Fixed in `0984d65`: added `greptile_reviewed_sha=$(cut -d: -f3)` and a guard that skips if the cached SHA doesn't match the live `head_sha`. Also updated the fake-gh test fixture to return empty output for `--jq` calls (matching real behaviour), and added a regression test for the stale-SHA scenario.

File: scripts/github/activity-gate.sh [hunk 2]
```diff
@@ -694,6 +694,93 @@ check_greptile_scores() {
...
+# generate a review_requested/mention notification. check_greptile_scores seeds
+# (and does NOT emit) on first discovery to avoid flooding when a repo is first
+# onboarded — so a new PR's first Greptile result is silently recorded and only
+# re-emitted after the 1-hour cooldown. This function closes that gap by emitting
+# on first discovery of a changed signature for bot-authored PRs.
+#
+# Routing:
+#   greptile < 4 (significant findings)   → greptile_needs_fix
+#   greptile = 4 (minor improvements)     → greptile_needs_improvement
+#   greptile >= 5 + CLEAN/MERGEABLE       → skip (check_merge_ready handles this)
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply scripts/github/activity-gate.sh]: The routing comment in the current code (after 16122e8) no longer mentions CLEAN/MERGEABLE — it only lists DIRTY/CONFLICTING and BLOCKED/UNKNOWN as skip cases, with greptile >= 5 as a standalone skip. The CLEAN condition was removed as part of the score=5/BEHIND fix. This finding no longer applies to the current code.

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: scripts/github/activity-gate.sh [hunk 3]
```diff
@@ -694,6 +694,93 @@ check_greptile_scores() {
...
+            greptile_score=$(cut -d: -f1 < "$greptile_state_file")
+        fi
+
+        # No Greptile review on file yet — skip
+        [ -z "$greptile_score" ] || [ "$greptile_score" = "null" ] && continue
+
+        # Greptile 5/5 + CLEAN/MERGEABLE → check_merge_ready already covers this
+        if [ "$greptile_score" -ge 5 ] 2>/dev/null && [ "$merge_state" = "CLEAN" ]; then
+            continue
+        fi
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply scripts/github/activity-gate.sh]: Fixed in `16122e8`: the `CLEAN` conjunct was removed from the score-≥-5 skip guard. The check now reads `if [ "$greptile_score" -ge 5 ]` — it skips whenever Greptile gives a perfect review regardless of `mergeStateStatus`. The other checks (`check_merge_ready`, `check_ci_failures`) own merge-state concerns.
